realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -70,6 +70,9 @@ JNIEXPORT jstring JNICALL Java_io_realm_internal_TableQuery_nativeValidateQuery
 #define RELEASE_ARRAY() \
     env->ReleaseLongArrayElements(columnIndexes, arr, 0);
 
+#define ABORT_ARRAY() \
+    env->ReleaseLongArrayElements(columnIndexes, arr, JNI_ABORT);
+
```

Comment:
0 copy back the content and free the elems buffer
JNI_COMMIT copy back the content but do not free the elems buffer
JNI_ABORT free the buffer without copying back the possible changes

I don't think even for `RELEASE_ARRAY` we need to copy anything back. Shall we just change the `RELEASE_ARRAY` to use `JNI_ABORT`?